pkg/certificate/providers/certmanager/certificate_manager_test.go

```diff
@@ -120,5 +119,135 @@ var _ = Describe("Test cert-manager Certificate Manager", func() {
 			Expect(getCertificateError).ToNot(HaveOccurred())
 			Expect(cachedCert).To(Equal(cert))
 		})
+
+		It("should rotate the certificate", func() {
+			mockConfigurator.EXPECT().GetServiceCertValidityPeriod().Return(validity).AnyTimes()
+
+			cert, err := cm.RotateCertificate(cn)
+			Expect(err).Should(BeNil())
+			cachedCert, err := cm.GetCertificate(cn)
+			Expect(cachedCert).To(Equal(cert))
+			Expect(err).Should(BeNil())
+		})
 	})
 })
+
+func TestReleaseCertificate(t *testing.T) {
+	cert := &Certificate{
+		commonName: cn,
+		expiration: time.Now().Add(1 * time.Hour),
+	}
+	manager := &CertManager{cache: map[certificate.CommonName]certificate.Certificater{cn: cert}}
+
+	testCases := []struct {
+		name       string
+		commonName certificate.CommonName
+	}{
+		{
+			name:       "release existing certificate",
```

Comment:
These two test cases are actually the same. Are we missing something here? Do we have a case that releasing cert is successful?

Reply:
The first one is testing releasing the certificate. The cert is released so then `manager.GetCertificate()` errors because the cert is no longer there.
The second test case is releasing the certificate that is no longer there.